src/cargo/sources/registry/remote.rs

```diff
@@ -179,8 +179,10 @@ impl<'cfg> RegistryData for RemoteRegistry<'cfg> {
         }
         // Check if the cache is valid.
         let current_version = self.current_version();
-        if current_version.is_some() && current_version.as_deref() == index_version {
-            return Poll::Ready(Ok(LoadResponse::CacheValid));
+        if let (Some(c), Some(i)) = (current_version, index_version) {
+            if c.ends_with(i) {
```

Comment:
Sorry. I am confused. Isn't `current_version` the git SHA for remote registry? I thought it would look like

- `index_version` -> `<file>:<remote>`
- `current_version` -> `<remote>`

So the code should be like:

```suggestion
            if i.ends_with(c.as_str()) {
```

I might be wrong, though.